### ISSUE TITLE
What is array_latitude_rad ?

### ISSUE THREAD

**Issue by torrance** [MWATelescope/mwa_hyperbeam#4]
I found it a little confusing what `array_latitude_rad` is meant to be, since it seems like it should be a scalar value but it's being passed in as a pointer to an array.

My guess is: 1. this is an array of size 1 (i.e. a pointer to a single double value) and 2. you're using a pointer so that the NULL pointer can double as 'no value' (since 0, -1, etc. are legitimate latitude values).

It might help to say this in the header file so it's clearer what's expected (or if I'm wrong, with the correct info!).

**Comment by cjordan** [comment]
Hi @torrance. I've downloaded the header to have a look, and this is the associated comment:

```
 * * `array_latitude_rad` - A pointer to an array latitude to use for the
 *   parallactic-angle correction. If the pointer is null, no correction is
 *   done.
```

I guess it's not hugely clear from this text that the PA correction is optional. The intention is to only apply it if the array latitude is given; that's why it's supplied via a pointer, so that the pointer is allowed to be null. Although the very top of the header comment for the `calc_jones` functions states:

```
 * Get the beam response Jones matrix for the given direction and pointing. Can
 * optionally re-define the X and Y polarisations and apply a parallactic-angle
 * correction; see
 * <https://github.com/MWATelescope/mwa_hyperbeam/blob/main/fee_pols.pdf>
```

There's maybe a little more info in the examples: https://github.com/MWATelescope/mwa_hyperbeam/blob/3f883cf293773e4ddf78f72f15fcd3bb2f467ae2/examples/fee.c#L52-L55

With all this stated, my question then is, what would you like to see to clear up your confusion?

**Comment by torrance** [comment]
The example was good and helped clarify what was expected.

In terms of the header, the use of the word 'array' is what I found so confusing. It communicated to me that it could be NULL just fine, but I was left wondering... how long is this array, and what kind of extra values does it expect beyond  the latitude value. The use of `array` versus simply `pointer` implies something here that is not meant.

I would rename the variable to `latitude_rad_ptr` or just `latitude_rad`, and clarify the help text as:

```
 * * `latitude_rad` - A pointer to the latitude to use for the
 *   parallactic-angle correction. If the pointer is null, no correction is
 *   done.
 ```

**Comment by cjordan** [comment]
I see, yes that is confusing. I've used "array" arround the place when it could be "telescope" instead. But, I'll go with `latitude_rad` and clarify that this is an instrument thing in the docs. A new release will include this within a couple of weeks.

**Comment by torrance** [comment]
Oh! you meant array as in MWA _array_? I see now - no I totally missed that meaning altogether!

**Comment by cjordan** [comment]
Addressed by 00e5549d0687b62cb0bf6cbb7bd25c37d245ab04
